src/core/main.cpp

```diff
@@ -567,6 +567,72 @@ int main( int argc, char * * argv )
 		}
 	}
 
+	// Test file argument before continuing
+
+	QFile testFile( fileToLoad );
+	if( testFile.open( QIODevice::ReadOnly) )
+	{
+		// Test if render ( coreOnly ) and file is empty.
+		if( coreOnly == true && testFile.size() == 0 )
+		{
+			// is empty
+			printf( "The file %s is empty.\n",
+					fileToLoad.toStdString().c_str() );
+			exit( 1 );
+		}
+		testFile.close();
+	}
+
+	if( !fileToLoad.isEmpty() )
+	{
+		QFileInfo fileInfoLoad( fileToLoad );
+		if( !fileInfoLoad.exists() )
+		{
+			printf( "The file %s does not exist.\n",
+					fileToLoad.toStdString().c_str() );
+			exit( 1 );
+		}
+		else if( fileInfoLoad.isDir() )
+		{
+			printf( "%s is a directory.\n",
+					fileToLoad.toStdString().c_str() );
+			exit( 1 );
+		}
+		else if( !(	fileInfoLoad.suffix().toLower() == "mmp" ||
+				fileInfoLoad.suffix().toLower() == "mmpz" ||
+				fileInfoLoad.suffix().toLower() == "mpt" ||
+				fileInfoLoad.suffix().toLower() == "xml" ) )
+		{
+			printf( "%s is not an LMMS project file.\n",
+					fileToLoad.toStdString().c_str() );
+			exit( 1 );
+		}
+	}
+	else if( !fileToImport.isEmpty() )
+	{
+		QFileInfo fileInfoImport( fileToImport );
+		if( !fileInfoImport.exists() )
+		{
+			printf( "The file %s does not exist.\n",
+					fileToImport.toStdString().c_str() );
+			exit( 1 );
+		}
+		else if( fileInfoImport.isDir() )
+		{
+			printf( "%s is a directory.\n",
+					fileToImport.toStdString().c_str() );
+			exit( 1 );
+		}
+		else if( !(	fileInfoImport.suffix().toLower() == "mid"	||
+				fileInfoImport.suffix().toLower() == "midi"	||
+				fileInfoImport.suffix().toLower() == "rmi"	||
+				fileInfoImport.suffix().toLower() == "h2song" ) )
```

Comment:
The import plug-ins know better how to handle these files; this check should be removed. `ImportFilter::import()` takes care of this.